### PR TITLE
Pass hashBits, invertHash to OneHotHashEncodingEstimator

### LINKED ISSUE #1560: Categorical Hash transform Create method ignores the number of HashBits
Looking at the [Create method](https://github.com/dotnet/machinelearning/blob/850f91c2e67679a825b49d3eefd96dea2cc2c153/src/Microsoft.ML.Transforms/CategoricalHashTransform.cs#L148) it doesn't pass the number of HashBits to the OneHotHashEncodingEstimator.

### PR DESCRIPTION
**Pull request by feiyun0112**
Fix #1560

### CODE REVIEW

File: src/Microsoft.ML.Transforms/CategoricalHashTransform.cs [hunk 1]
```diff
@@ -240,10 +240,16 @@ public sealed class ColumnInfo
...
+        /// <param name="hashBits">Number of bits to hash into. Must be between 1 and 30, inclusive.</param>
+        /// <param name="invertHash">Limit the number of keys used to generate the slot name to this many. 0 means no invert hashing, -1 means no limit.</param>
         /// <param name="outputKind">The type of output expected.</param>
-        public OneHotHashEncodingEstimator(IHostEnvironment env, string inputColumn,
-            string outputColumn = null, CategoricalTransform.OutputKind outputKind = Defaults.OutputKind)
-            : this(env, new ColumnInfo(inputColumn, outputColumn ?? inputColumn, outputKind))
+        public OneHotHashEncodingEstimator(IHostEnvironment env,
+            string inputColumn,
+            string outputColumn,
+            int hashBits,
```
Comment (Zruty0) [inline src/Microsoft.ML.Transforms/CategoricalHashTransform.cs]: >hashBits [](start = 16, length = 8)

could we have defaults here?

Comment (Zruty0) [reply src/Microsoft.ML.Transforms/CategoricalHashTransform.cs]: here and `invertHash`

---
In reply to: [231604796](https://github.com/dotnet/machinelearning/pull/1564#discussion_r231604796) [](ancestors = 231604796)

**Comment by Zruty0** [comment]
Thanks! Please address the comments, and then it's good to go

---
In reply to: [436707664](https://github.com/dotnet/machinelearning/pull/1564#issuecomment-436707664) [](ancestors = 436707664)

File: src/Microsoft.ML.Transforms/CategoricalCatalog.cs [hunk 2]
```diff
@@ -43,13 +43,17 @@ public static class CategoricalCatalog
...
         /// <param name="inputColumn">The input column</param>
         /// <param name="outputColumn">The output column. If <c>null</c>, <paramref name="inputColumn"/> is used.</param>
+        /// <param name="hashBits">Number of bits to hash into. Must be between 1 and 30, inclusive.</param>
+        /// <param name="invertHash">Limit the number of keys used to generate the slot name to this many. 0 means no invert hashing, -1 means no limit.</param>
         /// <param name="outputKind">The conversion mode.</param>
         /// <returns></returns>
         public static OneHotHashEncodingEstimator OneHotHashEncoding(this TransformsCatalog.CategoricalTransforms catalog,
                 string inputColumn,
                 string outputColumn = null,
+                int hashBits = OneHotHashEncodingEstimator.Defaults.HashBits,
```
Comment (Zruty0) [inline src/Microsoft.ML.Transforms/CategoricalCatalog.cs]: >hashBits [](start = 20, length = 8)

Could you please add a test to `test\Microsoft.ML.Tests\Transformers\CategoricalTests.cs` for this constructor?

**Comment by Zruty0** [comment]
@feiyun0112 sorry for the late review. It looks good to me now, but there are conflicts. Could you please merge with master, and then I'll merge it in?

Also please mention @Zruty0 so that I don't forget again :)

**Comment by sfilipi** [comment]
Hi @feiyun0112, still planning to wrap this?

**Comment by feiyun0112** [comment]
sorry for the late response. i have merged

**Pull request merged**